pywr-core/src/parameters/py.rs

```diff
@@ -120,7 +121,7 @@ impl Parameter for PyParameter {
                 py,
                 timestep.date.year(),
                 timestep.date.month() as u8,
-                timestep.date.day(),
+                timestep.date.day() as u8,
```

Comment:
I think we can use the `chrono` feature in pyo3 to pass the date directly as an argument.